modules/asm/scripts/install_asm.sh

```diff
@@ -177,15 +179,45 @@ else
     mkdir -p "${OUTDIR}"
 fi
 
-if [[ "${CA}" = "citadel" ]]; then
-    CA_COMMAND_SNIPPET="--ca citadel --ca_cert ${CA_CERT} --ca_key ${CA_KEY} --root_cert ${ROOT_CERT} --cert_chain ${CERT_CHAIN}"
+if [[ "${CA}" == "citadel" ]]; then
+    CA_COMMAND_SNIPPET="--ca citadel"
 else
     CA_COMMAND_SNIPPET=""
 fi
 
+if [[  "${CA_CERT}" != "none" ]]; then
```

Comment:
>  I think that is still needed, because you want to be able to run just the '--ca citadel' flag or the flag and the ca_certs maps or no --ca flag to use meshca.

IIUC all 4 `"ca_cert", "ca_key", "root_cert", "cert_chain"` will either always exist or none will as TF validation will guard this. In that case why do we need to split it up and have different conditionals for each flag?  There will either always be `--ca_key, --ca_key, --root_cert, --cert-chain` or none of them?

Reply:
@bharathkkb I updated the script. Please let me know if that is correct now.